streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/BaseJoinProcessorNode.java

```diff
@@ -1,89 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements. See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License. You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.kafka.streams.kstream.internals.graph;
-
-import org.apache.kafka.streams.kstream.ValueJoiner;
-
-/**
- * Utility base class containing the common fields between
- * a Stream-Stream join and a Table-Table join
- */
-abstract class BaseJoinProcessorNode<K, V1, V2, VR> extends StreamsGraphNode {
```

Comment:
Adding the extra types introduced some unresolvable differences between Stream and Table joins, so I just inlined this abstract class into the two subclasses.

It actually turns out that neither of the subclasses needed all of the information in here, so it wasn't a perfect superclass anyway.